### PR TITLE
[clang][analyzer] Improve 'errno' modeling of 'mkdtemp'

### PR DESCRIPTION
**Pull request by benshi001**

### CODE REVIEW

**Review by steakhal — APPROVED** [review]
LGTM

**Comment by benshi001** [comment]
> LGTM

Thanks for your help!

File: clang/test/Analysis/errno-stdlibraryfunctions.c [hunk 1]
```diff
@@ -54,3 +51,25 @@ int errno_lseek(int fildes, off_t offset) {
...
+    if (errno) {}                    // no warning
+  }
+}
+
+void errno_mkdtemp(char *template) {
+  char *Dir = mkdtemp(template);
+  if (Dir == NULL) {
+    clang_analyzer_eval(errno != 0); // expected-warning{{TRUE}}
+    if (errno) {}                    // no warning
+  } else {
```
Comment (balazske) [inline clang/test/Analysis/errno-stdlibraryfunctions.c:72]:
```suggestion
  } else {
    `clang_analyzer_eval(Dir == template); // expected-warning{{TRUE}}`
```

File: clang/lib/StaticAnalyzer/Checkers/StdLibraryFunctionsChecker.cpp [hunk 2]
```diff
@@ -2511,10 +2511,12 @@ void StdLibraryFunctionsChecker::initFunctionSummaries(
             .ArgConstraint(NotNull(ArgNo(0))));
 
     // char *mkdtemp(char *template);
-    // FIXME: Improve for errno modeling.
     addToFunctionSummaryMap(
         "mkdtemp", Signature(ArgTypes{CharPtrTy}, RetType{CharPtrTy}),
-        Summary(NoEvalCall).ArgConstraint(NotNull(ArgNo(0))));
+        Summary(NoEvalCall)
+            .Case({NotNull(Ret)}, ErrnoMustNotBeChecked, GenericSuccessMsg)
```
[the remark this answers is not in the document]
Comment (benshi001) [reply clang/lib/StaticAnalyzer/Checkers/StdLibraryFunctionsChecker.cpp]: Fixed. Thanks.

**Review by balazske — APPROVED** [review]
LGTM (if the tests pass)

**Comment by steakhal** [comment]
No release notes?

**Comment by benshi001** [comment]
> No release notes?

Sorry, I forget to pull your modification to my local repo, I will fix this soon.

**Comment by benshi001** [comment]
> No release notes?

I have created another PR to add the miss release note. https://github.com/llvm/llvm-project/pull/76805

**Pull request merged**